src/code/stores/graph-store.coffee

```diff
@@ -146,9 +146,31 @@ GraphStore  = Reflux.createStore
     @_graphUpdated()
     @updateListeners()
 
+  isUniqueTitle: (title, skipNode, nodes=@getNodes()) ->
+    nonUniqueNode = (otherNode) ->
+      sameTitle = otherNode.title is title
+      if skipNode then sameTitle and otherNode isnt skipNode else sameTitle
+    not _.find nodes, nonUniqueNode
+
+  ensureUniqueTitle: (node, newTitle=node.title) ->
+    nodes = @getNodes()
+    defaultTitle = tr "~NODE.UNTITLED"
+    if newTitle is defaultTitle or not @isUniqueTitle newTitle, node, nodes
```

Comment:
Good catch.  I changed the default index from 1 to 2 at the last minute and didn't retest with a blank canvas.  I'll update and test it without the default check.